File: NeetCode/Group_Anagrams.cpp
// class Solution {
// public:
//     vector<vector<string>> groupAnagrams(vector<string>& strs) {
//         unordered_map<string, vector<string>> m;
//         vector<vector<string>> result;
//         for(const auto& str:strs){
//             string sortedStr {str};
//             ranges::sort(sortedStr);
//             m[sortedStr].push_back(str);
//         }
//         for(auto& [_, x]: m)
//             result.push_back(move(x));
//             return result;
//     }
// };
// Time Complexity: O(n*mlogm), where n is the length of the array and m is the length of the longest string in the array.
// Space Complexity: O(n*m), where n is the length of the array and m is the length of the longest string in the array.

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result=[]
        table={}

        for each in strs:
            key=''.join(sorted(each))
            if key not in table:
                table[key]=[]
            table[key].append(each)
        for value in table.values():
            result.append(value)
        return result
        